collector/examples/collect.rs

```diff
@@ -259,84 +294,113 @@ struct Options {
+        help_heading = "Collect Request Parameters (Fixed Size)",
+    )]
+    batch_id: Option<BatchId>,
+}
 
-    LogTracer::init()?;
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    install_tracing_subscriber()?;
 
-    Ok(())
-}
+    let mut command = Options::command();
+    let mut matches = match command.try_get_matches_from_mut(std::env::args_os()) {
+        Ok(matches) => matches,
+        Err(err) => err.format(&mut command).exit(),
+    };
+    let options = match Options::from_arg_matches_mut(&mut matches) {
+        Ok(options) => options,
+        Err(err) => err.format(&mut command).exit(),
+    };
 
-async fn run_collection_generic<V: vdaf::Collector>(
-    parameters: CollectorParameters,
-    vdaf: V,
-    http_client: reqwest::Client,
-    interval: Interval,
-    agg_param: &V::AggregationParam,
-) -> Result<(), janus_collector::Error>
-where
-    for<'a> Vec<u8>: From<&'a V::AggregateShare>,
-    V::AggregateResult: Debug,
-{
-    let collector = Collector::new(parameters, vdaf, http_client);
-    let collection = collector.collect(interval, agg_param).await?;
-    println!("Aggregation result: {:?}", collection.aggregate_result());
-    println!("Number of reports: {}", collection.report_count());
-    Ok(())
+    match run(options).await {
+        Ok(()) => Ok(()),
+        Err(Error::Anyhow(err)) => Err(err),
+        Err(Error::Clap(err)) => err.format(&mut command).exit(),
+    }
 }
 
 // This function is broken out from `main()` for the sake of testing its argument handling.
 async fn run(options: Options) -> Result<(), Error> {
+    match (
+        &options.batch_interval_start,
+        &options.batch_interval_duration,
+        &options.batch_id,
+    ) {
+        (Some(batch_interval_start), Some(batch_interval_duration), None) => {
+            let batch_interval = Interval::new(
+                Time::from_seconds_since_epoch(*batch_interval_start),
+                Duration::from_seconds(*batch_interval_duration),
+            )
+            .map_err(|err| Error::Anyhow(err.into()))?;
+            run_with_query(options, Query::new_time_interval(batch_interval)).await
+        }
+        (None, None, Some(batch_id)) => {
+            let batch_id = *batch_id;
+            run_with_query(options, Query::new_fixed_size(batch_id)).await
+        }
+        (None, None, None) | (Some(_), None, None) | (None, Some(_), None) => {
+            Err(clap::Error::raw(
+                ErrorKind::MissingRequiredArgument,
+                "missing collect request parameters".to_string(),
+            )
+            .into())
+        }
+        (Some(_), Some(_), Some(_)) | (Some(_), None, Some(_)) | (None, Some(_), Some(_)) => {
+            Err(clap::Error::raw(
+                ErrorKind::ArgumentConflict,
+                "time interval and fixed size collect request parameters cannot be combined"
+                    .to_string(),
+            )
+            .into())
+        }
```

Comment:
If the `batch_interval_start`, `batch_interval_duration` and `batch_id`  args are appropriately marked with [`Arg::requires`](https://docs.rs/clap/latest/clap/struct.Arg.html#method.requires) and `conflicts`, could we replace these with a single case matching `_` that just returns some generic error or panics, on the premise that `clap` will have caught incorrect arguments before we get here?

Reply:
That was a nice cleanup, thanks!